esphome/components/lvgl/rotary_encoders.py

```diff
@@ -0,0 +1,62 @@
+import esphome.codegen as cg
+from esphome.components.binary_sensor import BinarySensor
+from esphome.components.rotary_encoder.sensor import RotaryEncoderSensor
+import esphome.config_validation as cv
+from esphome.const import CONF_GROUP, CONF_ID, CONF_SENSOR
+
+from .defines import (
+    CONF_ENTER_BUTTON,
+    CONF_LEFT_BUTTON,
+    CONF_LONG_PRESS_REPEAT_TIME,
+    CONF_LONG_PRESS_TIME,
+    CONF_RIGHT_BUTTON,
+    CONF_ROTARY_ENCODERS,
+)
+from .helpers import lvgl_components_required
+from .lvcode import add_group, lv, lv_add, lv_expr
+from .schemas import ENCODER_SCHEMA
+from .types import lv_indev_type_t
+
+ROTARY_ENCODER_CONFIG = cv.ensure_list(
+    ENCODER_SCHEMA.extend(
+        {
+            cv.Required(CONF_ENTER_BUTTON): cv.use_id(BinarySensor),
+            cv.Required(CONF_SENSOR): cv.Any(
+                cv.use_id(RotaryEncoderSensor),
+                cv.Schema(
+                    {
+                        cv.Optional(CONF_LEFT_BUTTON): cv.use_id(BinarySensor),
+                        cv.Optional(CONF_RIGHT_BUTTON): cv.use_id(BinarySensor),
```

Comment:
Should these be required?
If not, the `to_code` will also need to check if they have been set in config or it will fail on a `KeyError`
```suggestion
                        cv.Required(CONF_LEFT_BUTTON): cv.use_id(BinarySensor),
                        cv.Required(CONF_RIGHT_BUTTON): cv.use_id(BinarySensor),
```